From 8ccda67643b41ecbaf989116d31520d70c726db1 Mon Sep 17 00:00:00 2001
From: Kirill Kapranov <kirill.kapranov@compulab.co.il>
Date: Fri, 31 Mar 2023 21:38:32 +0300
Subject: [PATCH 87/89] iio:max11108: Fix name of trigger to be allocated

On probe method the ADC driver allocates a trigger. Allocation of the
trigger with a constant name restricts the system to use  only one ADC.
In order to allow usage of more than one ADC, the trigger name has been
suffixed with an unique ID, per sample of other drivers.

Signed-off-by: Kirill Kapranov <kirill.kapranov@compulab.co.il>
---
 drivers/iio/adc/max11108.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/iio/adc/max11108.c b/drivers/iio/adc/max11108.c
index 60b740c76343..dfb115d2679f 100644
--- a/drivers/iio/adc/max11108.c
+++ b/drivers/iio/adc/max11108.c
@@ -183,7 +183,8 @@ static int max11108_probe(struct spi_device *spi)
 		return ret;
 	}
 	/* Create default trigger */
-	state->trig = devm_iio_trigger_alloc(&spi->dev, "max11108-trigger");
+	state->trig = devm_iio_trigger_alloc(&spi->dev, "%s-dev%d",
+			indio_dev->name, iio_device_id(indio_dev));
 	if (state->trig == NULL) {
 		ret = -ENOMEM;
 		dev_err(&indio_dev->dev, "Failed to allocate iio trigger\n");
-- 
2.17.1

